### PR TITLE
Downlink timing fix

### PR DESCRIPTION
**Pull request by wero1414**
I modified the lora_cycle to get the standard timing for class C and Class A 

Already tested on Class C, working with a Bug that i wasnt able to catch today, this bug happens when you do the first downlink to the device, once it have received it when the function LORA_Receive_Data() is called, the function RFM_Get_Package() is called inside, and it always return a good CRC, so it gets like if it where a new packet, BUT right now with this example im able to send donwlinks in class C while sending uplinks or when not sending uplinks.

Any help reproducing the bug and also trying to fix it will be really handy!

Do not merge yet, it need more changes! like the support for different regions

### CODE REVIEW

**Comment by wero1414** [comment]
UPDATE

Class A downlink working stable

**Comment by mr3188** [comment]
@wero1414 I'm facing similar bug in my fix, once I send a confirmed downlink then node detect's it but the ackpending flag remains set even if I reset it.

The debug session shows subsequent downlinks are detected but lost somewhere in de code and de bugger jumps from one place to another without sense. I will discard my changes and begin from your branch to see if I can fix it and test it in EU868.

**Comment by mr3188** [comment]
@wero1414  don't worry, I was stuck.

**Comment by wero1414** [comment]
Just add an example for Serial to LoRaWAN with Class C with ABP

**Comment by mr3188** [comment]
@wero1414  Is this ready for merge? Do you still need help with the bug?

I have a branch with some basic  MAC command features including LinkCheck working but it is based on the master. If the work in this branch is finished and just waiting to be merged I can easily modify mine to be based on this. If not, I can try to reproduce the bug and test it with EU_868.

I have started to work on ADR but as much of the work to do is in LoRaMAC.cpp I want to be sure I'm working on the correct branch.

**Comment by wero1414** [comment]
Well i havent been able to find a solution for the BUG and also this branch needs the different configurations for different regions, thats all it needs

**Comment by mr3188** [comment]
Ok, if you don't mind I will take a look at EU868 changes needed to try to solve the bug.

**Comment by mr3188** [comment]
In RF95.cpp LORA_Receive_Data() always set MessageStatus=NEW_MESSAGE if node class is not Class A and then calls RFM_Get_Package() without checking RxDone, then CRC Error interrupt is check but as the Interrupt register returns 0 (no interrupt flagged) it always return CRC_OK and  read 0 bytes from Fifo.
I'm still checking because I can't receive downlinks I always get crc error. I will compare the code with the version I used to create MAX_DONLINK_PAYLOAD_SIZE but may be this information gives you a clue to fix the issue as you know this branch better than me.

File: src/arduino-rfm/LoRaMAC.cpp [hunk 1]
```diff
@@ -67,60 +67,76 @@ void LORA_Cycle(sBuffer *Data_Tx, sBuffer *Data_Rx, RFM_command_t *RFM_Command,
...
-  {
-    //Lora send data
-    LORA_Send_Data(Data_Tx, Session_Data, LoRa_Settings);
-	prevTime = millis();
+	unsigned char rx1_ch = LoRa_Settings->Channel_Rx;
+	unsigned char rx1_dr = LoRa_Settings->Datarate_Tx+10;
+  	//Transmit
+	if(*RFM_Command == NEW_RFM_COMMAND){
+		#if (SAMR34)
+		pinMode(RFM_SWITCH,OUTPUT);
```
Comment (sabas1080) [inline src/arduino-rfm/LoRaMAC.cpp:75]: creo que RFM_Switch debería agregarse a sRFM_pins para que este disponible para cualquier placa

File: src/arduino-rfm/RFM95.cpp [hunk 2]
```diff
@@ -706,6 +701,12 @@ void RFM_Continuous_Receive(sSettings *LoRa_Settings)
   RFM_Change_Datarate(SF12BW125);
   RFM_Change_Channel(CHRX2);
 #else
+  //Datarate for downlink should be 8 but testing on 10
+  //LoRa_Settings->Datarate_Rx=10;
```
Comment (sabas1080) [inline src/arduino-rfm/RFM95.cpp:705]: Deberian eliminarse estos comentarios?

File: src/arduino-rfm/RFM95.cpp [hunk 3]
```diff
@@ -706,6 +701,12 @@ void RFM_Continuous_Receive(sSettings *LoRa_Settings)
...
+  //LoRa_Settings->Channel_Rx=0;
+  //Serial.println("DataRate Rx "+String(LoRa_Settings->Datarate_Rx));
+  //Serial.println("Cannel Rx "+String(LoRa_Settings->Channel_Rx));
+
```
Comment (sabas1080) [inline src/arduino-rfm/RFM95.cpp:709]: Y estos comentarios tambien deberian eliminarse?

**Review by sabas1080 — CHANGES_REQUESTED** [review]

**Review by ivanmorenoj — COMMENTED** [review]
Seems good to me.

**Comment by wero1414** [comment]
update timming on Rx for US915

**Pull request merged**